lib/kadalulib.py

```diff
@@ -87,23 +89,64 @@ def is_server_pod_reachable(hosts, port=24007, timeout=20):
     Returns False server pods are not reachable even after the timeout.
     """
 
-    for host in hosts:
-        retry_count = 0
-        while retry_count < 4:
+    connected = False
+    retry_count = 0
+
+    while retry_count < 4:
+        sockets = []
+
+        for host in hosts:
+            connect_started = False
             try:
-                with socket.create_connection((host, int(port)), timeout=timeout) as sock:
-                    sock.shutdown(socket.SHUT_RDWR)
-                return True
-            except socket.error:
-                logging.info(logf(
-                    "Waiting for the server pod to come up...",
-                    server_pod=host,
-                    retry_count=retry_count+1
+                sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
+                sock.setblocking(False)
+                sock.connect((host, int(port)))
+
+                connect_started = True
+
+            except socket.error as e:
+                if e.errno == errno.EINPROGRESS:
+                    connect_started = True
+                else:
+                    logging.info(logf(
+                        "Error encountered for server pod...",
+                        error=e,
+                        server_pod=host,
+                        retry_count=retry_count+1
+                    ))
+
+            if connect_started:
+                sockets.append(sock)
+
+        # Only care about output sockets (not input or exception)
+        readable, writable, exceptional = select.select([], sockets, [], timeout)
+
+        if not (readable or writable or exceptional):
```

Comment:
```suggestion
        # Only care about output sockets (not input or exception)
        _, writable, _ = select.select([], sockets, [], timeout)

        if not writable:
```
as ntg is supplied to select for rlist & xlist, the returned values will be of no significance afaik and could be discared, wdyt?

Reply:
Yes, I agree that would be better.